### PR TITLE
Add SchemaSchema, validate all schemas

### PR DESCRIPTION
**Pull request by lavalamp**
Basically no bugs found and coverage is up a tad.

Possibly the most self-referential thing I've ever written?

In unrelated news, some error messages have been improved a touch.

### CODE REVIEW

File: typed/validate_test.go [hunk 1]
```diff
@@ -26,6 +27,15 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
+func schemaSchema() *schema.Schema {
```
Comment (apelisse) [inline typed/validate_test.go]: This is not used

**Comment by apelisse** [comment]
I'll let you decide if you want to remove the unused function now or in a follow up
/lgtm
/hold

File: typed/validate_test.go (hunk 1 above)
Comment (lavalamp) [reply typed/validate_test.go]: fixed

**Comment by apelisse** [comment]
Thanks
/lgtm

**Pull request merged**